beefy-gadget/src/worker.rs

```diff
@@ -402,3 +384,133 @@ where
 
 	header.digest().convert_first(|l| l.try_to(id).and_then(filter))
 }
+
+/// Calculate the candidate block for the next BEEY vote.
+///
+/// Note, `number` is only used for better tracing output.
+fn vote_candidate<B>(
+	number: NumberFor<B>,
+	best_grandpa: NumberFor<B>,
+	best_beefy: NumberFor<B>,
+	min_delta: u32,
+) -> NumberFor<B>
+where
+	B: Block,
+{
+	let diff = best_grandpa.saturating_sub(best_beefy);
+	let diff = diff.saturated_into::<u32>();
```

Comment:
Note that the `candidate` must always be lower than `best_grandpa`. That was actually the reason we were doing `/ 2` in the previous code. Currently it will be possible to vote on `candidate` only iff `diff` is a power of two (as seen in the test cases).

Reply:
If we keep the `/ 2` we will never catch up is the delta between `best_grandpa` and `best_beefy` becomes larger than `min_delta`. Please see the previous commit and the tests.

The reason is that div 2 is basically the same as a minus 1 from the exponent. So we are calculating basically one power of two 'to less' such to speak.

Reply:
Makes sense! Some more context in https://github.com/paritytech/grandpa-bridge-gadget/issues/162